internal/auth0/organization/resource_connections.go

```diff
@@ -149,6 +163,12 @@ func updateOrganizationConnections(ctx context.Context, data *schema.ResourceDat
 
 	organizationID := data.Id()
 
+	connections := expandOrganizationConnections(data.GetRawConfig().GetAttr("enabled_connections"))
+	connectionMap := make(map[string]*management.OrganizationConnection)
+	for _, connection := range connections {
+		connectionMap[connection.GetConnectionID()] = connection
+	}
```

Comment:
We are already getting the correct IDs from the value.Difference, what makes us do this here?

Reply:
this is using the same code create uses to generate the OrganizationConnection, which is slightly better from a re-use point of view, and also fixed some errors I was getting

Reply:
The logic here isn't quite right, we already have the information that this expand gives us through the `toAdd` and the `toRemove` parameters. 

What was the error you were getting? 

We really don't need this part IMO.

Reply:
I am re-using the code in `expandOrganizationConnections` to map the data instead of doing the same thing in two different places. the map is just to access the ones we need

Reply:
the problem is that using the values in toAdd doesn't properly preserve nullness, which is necessary to work with our SDK

Reply:
Ah I see now, the `value.Difference` uses the `data *schema.ResourceData` and that one is having issues with bool values and knowing if they were set in the config or not. 

So we're expanding the connections so we can get the real bool attribute values (nil, false or true) before we add a connection. 

Let's make sure we add a comment on line 166 with this context, as otherwise we might risk forgetting :) when we come back to this.